core/src/View/Kokkos_ViewAccessPreconditionsCheck.hpp

```diff
@@ -0,0 +1,161 @@
+// the specified error message.
+template <class MemorySpace, class AccessSpace>
+struct RuntimeCheckBasicViewMemoryAccessViolation<MemorySpace, AccessSpace,
+                                                  false> {
+  KOKKOS_FUNCTION RuntimeCheckBasicViewMemoryAccessViolation(
+      Kokkos::Impl::SharedAllocationTracker const &tracker) {
+    char err[256] =
+        "Kokkos::View ERROR: attempt to access inaccessible memory space "
+        "(label=\"";
+
+    KOKKOS_IF_ON_HOST(({
+      if (tracker.has_record()) {
+        strncat(err, tracker.template get_label<void>().c_str(), 128);
+      } else {
+        strcat(err, "**UNMANAGED**");
+      }
+    }))
+
+    KOKKOS_IF_ON_DEVICE(({
+      strcat(err, "**UNAVAILABLE**");
+      (void)tracker;
+    }))
+
+    strcat(err, "\")");
+
+    Kokkos::abort(err);
+  }
+};
+
+template <class MemorySpace>
+KOKKOS_FUNCTION void runtime_check_memory_access_violation(
+    SharedAllocationTracker const &track) {
+  KOKKOS_IF_ON_HOST(((void)RuntimeCheckBasicViewMemoryAccessViolation<
+                         MemorySpace, DefaultHostExecutionSpace>(track);))
+  KOKKOS_IF_ON_DEVICE(((void)RuntimeCheckBasicViewMemoryAccessViolation<
+                           MemorySpace, DefaultExecutionSpace>(track);))
+}
+
+template <class IndexType, std::size_t... Extents, class... Indices,
+          std::size_t... Enumerate>
+KOKKOS_FUNCTION bool within_range(
+    Kokkos::extents<IndexType, Extents...> const &exts,
+    std::index_sequence<Enumerate...>, Indices... indices) {
+  // FIXME[CUDA11]: This is written so weirdly to avoid warnings with CUDA 11
+  // Without the workaround this could just be written as:
+  // return ((indices < exts.extent(Enumerate)) && ...) &&
+  //     ((std::is_unsigned_v<decltype(indices)> ||
+  //       (indices >= static_cast<decltype(indices)>(0))) &&
+  //      ...);
+  [[maybe_unused]] auto check_index_min = [](auto idx) {
+    if constexpr (!std::is_unsigned_v<decltype(idx)>) {
+      return idx >= static_cast<decltype(idx)>(0);
+    }
+
+    return true;
+  };
+
+  return ((indices < exts.extent(Enumerate)) && ...) &&
+         (check_index_min(indices) && ...);
+}
+
+template <class... Indices>
+KOKKOS_FUNCTION constexpr char *append_formatted_multidimensional_index(
+    char *dest, Indices... indices) {
+  char *d = dest;
+  strcat(d, "[");
+  (
+      [&] {
+        d += strlen(d);
+        to_chars_i(d,
+                   d + 20,  // 20 digits ought to be enough
+                   indices);
+        strcat(d, ",");
+      }(),
+      ...);
+  d[strlen(d) - 1] = ']';  // overwrite trailing comma
+  return dest;
+}
+
+template <class IndexType, size_t... Extents, std::size_t... Enumerate>
+KOKKOS_FUNCTION void print_extents(
+    char *dest, Kokkos::extents<IndexType, Extents...> const &exts,
+    std::index_sequence<Enumerate...>) {
+  append_formatted_multidimensional_index(dest, exts.extent(Enumerate)...);
+}
```

Comment:
This does not sound like an AccessPreconditionCheck to me. Is there no better place to put it?